Meta Powers screen prototype for easier debugging

nandastone opened the pull request:
This MR is a prototype for the Meta Powers screen as described in https://github.com/FreezingMoon/AncientBeast/issues/880

It allows two powers that I've found make debugging other issues significantly easier:

1. "Execution Mode" - turn on then click any creature to instantly kill it.
2. "Reset Cooldowns" - click to refresh player and creature cooldowns.

The Meta Powers screen can be accessed after starting a hotseat game in development mode, then pressing the Meta + P  hotkey. It can be closed using the same hotkey, or pressing ESC, or clicking the close icon.

![CleanShot 2021-12-06 at 16 06 01](https://user-images.githubusercontent.com/199204/144795870-f6fe12c9-9f71-4fbf-84a9-5186723ab36e.png)

Watch the feature in action: https://cln.sh/J8FWzs

To simplify sending commands like "Refresh Cooldown" throughout the app I've introduced a concept of event channels based on https://photonstorm.github.io/phaser-ce/Phaser.Signal.html#dispatch. 

This means less direct coupling between the classes, for example we do not need to pass the `MetaPowers()` instance into the `Ability()` class or anywhere else to correctly handle the new functionality.

I also refactored the Score screen CSS as it is shared by the Meta Powers screen.

![CleanShot 2021-12-06 at 16 06 43](https://user-images.githubusercontent.com/199204/144795936-b8e04176-4836-4fd6-b614-be7d48d17461.png)
![CleanShot 2021-12-06 at 16 06 36](https://user-images.githubusercontent.com/199204/144795943-ec1dacd3-2ad3-4da2-997a-bacabe0f2e69.png)

Caution: I have no tested these tools with the saved game logs. I personally think it's fine for those features to be exclusive.

<a href="https://gitpod.io/#https://github.com/FreezingMoon/AncientBeast/pull/1937"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

DreadKnight commented:
Pretty cool, I like that you're reusing those divs. Cursor shouldn't be clicking hand all around the MetaPowers screen btw.

Hotkey should change to Ctrl+Meta+P, seems that my desktop environment (KDE) has a default use on Meta+P that deals with external screens and it's kinda hard to change overall tbh. I'll have to make sure I mention this functionality in the contribution guide.

The tooltips could be more instantaneous, like we have have in the score screen for the flee button, but above, like https://game-icons.net does when you hover over those icons. Or like we have for the emojies in the score screen.

DreadKnight commented:
Also, not fully feeling that Reset Cooldowns icon, because it was first used for googling abilities and now planning to use it for something else (that #1728 issue that I've mentioned before). We should use https://game-icons.net/1x1/delapouite/extra-time.html

nandastone commented:
@DreadKnight

Not sure where the pointer cursor came from, maybe an interaction I'm not aware of. As far as I can tell it's no longer there:

https://user-images.githubusercontent.com/199204/144827337-143f6ec4-159f-464e-a16a-1caf71db7c40.mp4

Changed the hotkey to Ctrl  + Meta + P.

Added a description to `CONTRIBUTING.md`.

Swapped from native `title=""` tooltips to the instant tooltips:

![CleanShot 2021-12-06 at 20 06 18](https://user-images.githubusercontent.com/199204/144828143-18589893-43ed-4765-95e9-879932ca62bb.gif)

Updated the "Reset Cooldowns" icon:

![CleanShot 2021-12-06 at 20 13 43](https://user-images.githubusercontent.com/199204/144828239-70287839-60d3-43df-80dd-e40799f48d0f.png)

DreadKnight commented:
I usually test in Gitpod as it autodeploys and it's super convenient for me, but noticed that it usually messes some styling up a bit, like displaying health and plasma numbers from near the units somewhat below. Usually using 4 pixel black outline for the icons, configurable as well from their website to improve readability overall (though at times there are tiny glitches that I report).
![Screenshot_20211206_122746](https://user-images.githubusercontent.com/587487/144830389-bd2efe71-88dd-466c-9bf3-7a15eb2b4ec7.jpg)

Great job once again! 👏🏻 For some reason hotkey doesn't work in Gitcoin but will hopefully do so in the beta deployment xD
